### DISCUSSION TITLE
Total Storage usage per user.

### DISCUSSION THREAD

**Discussion by basicBrogrammer** [supabase/supabase#2673] [Questions] [answered]
Hey y'all I'm currently evaluating supabase for a project. One requirement would be to give the user a specific amount of upload storage for free. Then, the user can buy additional storage. 

Is it possible to calculate the total storage usage per user? 

Thanks!

**Comment by inian** [comment, marked as answer]
Can you try running this on your database @basicBrogrammer? `objects.owner` should have the owner who uploaded the files. So you can group by that field. 

```sql
select sum((metadata->>'size')::int), owner from storage.objects group by owner;
```

**Reply by basicBrogrammer** [reply]
Thanks for the response. Unfortunately, I haven't started up my app with supabase yet. I will be this week hopefully. I can try it then. Currently, I'm just looking through the docs. So storage objects are saved in the DB or is the `storage.objects` table just a pointer to something like S3?

**Reply by inian** [reply]
nice! The actual objects are stored in S3, and not in your database.
